File: backend/app/schemas/lead.py
import uuid
from datetime import datetime, date
from pydantic import BaseModel, ConfigDict


# ── 线索跟进记录 ─────────────────────────────────────────────
class LeadFollowUpCreate(BaseModel):
    follow_type: str
    content: str
    next_date: str | None = None


class LeadFollowUpOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    lead_id: uuid.UUID
    follow_type: str
    content: str
    next_date: str | None
    created_by: uuid.UUID | None
    created_at: datetime


# ── 线索 ─────────────────────────────────────────────────────
class LeadBase(BaseModel):
    company_name: str
    contact_name: str | None = None
    contact_phone: str
    contact_email: str | None = None
    source: str
    industry: str | None = None
    province: str | None = None
    city: str | None = None
    certification_interest: str | None = None
    status: str = "待跟进"
    assigned_to: uuid.UUID | None = None
    next_follow_up_date: date | None = None
    remark: str | None = None


class LeadCreate(LeadBase):
    pass


class LeadUpdate(BaseModel):
    company_name: str | None = None
    contact_name: str | None = None
    contact_phone: str | None = None
    contact_email: str | None = None
    source: str | None = None
    industry: str | None = None
    province: str | None = None
    city: str | None = None
    certification_interest: str | None = None
    status: str | None = None
    assigned_to: uuid.UUID | None = None
    next_follow_up_date: date | None = None
    remark: str | None = None


class LeadOut(LeadBase):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    created_by: uuid.UUID | None
    created_at: datetime
    updated_at: datetime


class LeadListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    company_name: str
    contact_name: str | None
    contact_phone: str
    source: str
    industry: str | None
    province: str | None
    city: str | None
    certification_interest: str | None
    status: str
    assigned_to: uuid.UUID | None
    next_follow_up_date: date | None
    created_at: datetime

File: frontend/src/api/lead.ts
import request from './request'

export interface LeadListItem {
  id: string
  company_name: string
  contact_name: string | null
  contact_phone: string
  source: string
  industry: string | null
  province: string | null
  city: string | null
  certification_interest: string | null
  status: string
  assigned_to: string | null
  next_follow_up_date: string | null
  created_at: string
}

export interface LeadDetail extends LeadListItem {
  contact_email: string | null
  remark: string | null
  updated_at: string
  created_by: string | null
}

export interface LeadFollowUp {
  id: string
  lead_id: string
  follow_type: string
  content: string
  next_date: string | null
  created_by: string | null
  created_at: string
}

export const leadApi = {
  list(params: { page?: number; page_size?: number; status?: string; keyword?: string }) {
    return request.get<any, { code: number; data: LeadListItem[]; total: number }>('/leads', { params })
  },
  get(id: string) {
    return request.get<any, { code: number; data: LeadDetail }>(`/leads/${id}`)
  },
  create(data: Record<string, unknown>) {
    return request.post<any, { code: number; data: LeadDetail }>('/leads', data)
  },
  update(id: string, data: Record<string, unknown>) {
    return request.put<any, { code: number; data: LeadDetail }>(`/leads/${id}`, data)
  },
  delete(id: string) {
    return request.delete(`/leads/${id}`)
  },
  listFollowUps(leadId: string) {
    return request.get<any, { code: number; data: LeadFollowUp[] }>(`/leads/${leadId}/follow-ups`)
  },
  addFollowUp(leadId: string, data: Record<string, unknown>) {
    return request.post(`/leads/${leadId}/follow-ups`, data)
  },
}
